core/i18n/src/ITwinLocalization.ts

```diff
@@ -51,25 +50,50 @@ export class ITwinLocalization implements Localization {
       caches: [],
       ...options?.detectorOptions,
     };
+
     this._initOptions = {
       interpolation: { escapeValue: true },
       fallbackLng: "en",
+      maxRetries: 1, // prevents wasted time and potential timeouts when requesting localization files throws an error
       backend: this._backendOptions,
       detection: this._detectionOptions,
       ...options?.initOptions,
     };
 
     this.i18next
       .use(options?.detectorPlugin ?? i18nextBrowserLanguageDetector)
-      .use(options?.backendPlugin ?? XHR)
+      .use(options?.backendPlugin ?? Backend)
       .use(TranslationLogger);
   }
 
   public async initialize(namespaces: string[]): Promise<void> {
+    let initNamespaces: readonly string[];
+    let combinedNamespaces: string[];
+    let defaultNS: string;
+
+    if (this._initOptions.ns === undefined) {
+      initNamespaces = []
+    } else if (typeof this._initOptions.ns === "string") {
+      initNamespaces = [this._initOptions.ns]
+    } else {  // === string[]
+      initNamespaces = this._initOptions.ns;
+    }
+    combinedNamespaces = [...new Set([...namespaces, ...initNamespaces])]; // without duplicates
+
+    defaultNS = combinedNamespaces[0];
```

Comment:
Maybe you could add the `defaultNS` above so you don't have to check for it at all at this point?

Reply:
If you decided against, this, another possibility is to keep the combinedNamespaces as a Set, then you don't have to check for the existence on line 88, just add it to the Set

Reply:
Good suggestions. I was able to clean this up a lot more by keeping `combinedNamespaces` as a Set and using the `??` operator.